components/event-service/internal/events/subscribed/events.go

```diff
@@ -36,13 +43,55 @@ type eventsClient struct {
 	triggerLister kneventinglister.TriggerLister
 }
 
+// waitForInformersSyncOrDie blocks until all informer caches are synced, or panics after a timeout.
+func waitForInformersSyncOrDie(f externalversions.SharedInformerFactory) {
+	log.Println("waiting for informers caches sync...")
+	ctx, cancel := context.WithTimeout(context.Background(), informerSyncTimeout)
+	defer cancel()
+
+	err := hasSynced(ctx, f.WaitForCacheSync)
+	if err != nil {
+		log.Fatalf("Error waiting for caches sync: %s", err)
```

Comment:
Is the call to ```log.Fatalf``` the best solution here?

Reply:
I think the idea is to die early if the sync didn't work.

Reply:
Yes, this was clear for me, but it implies a pod restarting. Should the issue be solved after this restart? And if not? As I remember, the informer has a "resync" logic, with a resync duration. Could this be used somehow?
It remembers me to STONITH, which should be the last solution... :)